sootup.analysis.interprocedural/src/main/java/sootup/analysis/interprocedural/icfg/JimpleBasedInterproceduralCFG.java

```diff
@@ -109,60 +108,74 @@ public Collection<Stmt> load(SootMethod method) {
 
   public JimpleBasedInterproceduralCFG(
       View view,
-      MethodSignature mainMethodSignature,
+      List<MethodSignature> entryPoints,
+      boolean enableExceptions,
+      boolean includeReflectiveCalls) {
+    this(
+        new ClassHierarchyAnalysisAlgorithm(view).initialize(entryPoints),
+        view,
+        entryPoints,
+        enableExceptions,
+        includeReflectiveCalls);
+  }
+
+  public JimpleBasedInterproceduralCFG(
+      CallGraph cg,
+      View view,
+      List<MethodSignature> entryPoints,
       boolean enableExceptions,
       boolean includeReflectiveCalls) {
     super(enableExceptions);
     this.includeReflectiveCalls = includeReflectiveCalls;
     this.view = view;
-    this.mainMethodSignature = mainMethodSignature;
-    cg = initCallGraph();
+    this.entryPoints = entryPoints;
```

Comment:
@kadirayk @JonasKlauke, should we allow users to pass entry points here to create jimple cfg? Earlier I was passing entry points (which I used to create cg) instead of actual cg.getEntrypoints() (which add clinit calls also). 
This created a lot of confusion for me and was giving me NullPointerExceptions, so should we just allow cg & use entry points from there?

Reply:
We don't need the entry points anymore if we get the CG from outside. But we should investigate the exceptions further.